File: ios/LanGaming/Sources/LanGaming/LanGamingListener.swift
import Foundation

public protocol LanGamingListener: AnyObject {
    func onPeerConnected(address: String, playerId: Int, displayName: String)
    func onPeerDisconnected(address: String)
    func onMessageReceived(address: String, messageType: UInt8, payload: Data)
    func onError(_ message: String)
}

public extension LanGamingListener {
    func onPeerConnected(address: String, playerId: Int, displayName: String) {}
    func onPeerDisconnected(address: String) {}
    func onMessageReceived(address: String, messageType: UInt8, payload: Data) {}
    func onError(_ message: String) {}
}

File: android/lan-gaming/src/main/kotlin/com/locxx/langaming/LanGamingListener.kt
package com.locxx.langaming

interface LanGamingListener {
    fun onPeerConnected(address: String, playerId: Int, displayName: String) {}
    fun onPeerDisconnected(address: String) {}
    fun onMessageReceived(address: String, messageType: Byte, payload: ByteArray) {}
    fun onError(message: String) {}
}
